app/components/elements/forms/nested_component.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Elements
+  module Forms
+    # Encapsulates a nested form, including adding and removing nested models.
+    class NestedComponent < ApplicationComponent
+      def initialize(form:, model_class:, field_name:, form_component:)
+        @form = form
+        @model_class = model_class
+        @field_name = field_name.to_sym
+        @form_component = form_component
+        super()
+      end
+
+      attr_reader :form, :model_class, :field_name, :form_component
+
+      def body_id
+        "card-body-#{field_name}"
+      end
+
+      def header_label
+        model_class.model_name.plural.titleize
+      end
+    end
+  end
+end
```

Comment:
Borrowed from POC and pared down a bit (no dedicated card component element yet).